src/main/src/main/java/com/shevtsov/dao/impl/ClientDaoImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.shevtsov.dao.impl;
+
+import com.shevtsov.dao.ClientDao;
+import com.shevtsov.domain.Client;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ClientDaoImpl implements ClientDao {
+    private Map<Long, Client> map = new HashMap<>();        //storage emulation
+    private static long generator = 0;
+    private static final ClientDao instance = new ClientDaoImpl();
+
+    private ClientDaoImpl() {
+    }
+
+    public static ClientDao getInstance() {
+        return instance;
+    }
+
+    @Override
+    public long save(Client client) {
+        client.setId(generator++);
```

Comment:
please move this to second method and simplefy this

Reply:
done